docs/plugins/ingest-attachment.asciidoc

```diff
@@ -0,0 +1,34 @@
+[[ingest-attachment]]
+== Ingest Attachment Processor Plugin
+
+The ingest attachment plugin lets Elasticsearch extract file attachments in common formats (such as PPT, XLS, PDF)
+using the Apache text extraction library http://lucene.apache.org/tika/[Tika].
+
+It can be used as replacement for the mapper attachment plugin.
+
+The source field must be a base64 encoded binary.
+
+[[ingest-attachment-options]]
+.Attachment options
+[options="header"]
+|======
+| Name                   | Required  | Default          | Description
+| `source_field`         | yes       | -                | The field to get the base64 encoded field from
+| `target_field`         | no        | attachment       | The field that will hold the attachment information
+| `indexed_chars`        | no        | Integer.MAX_INT  | The number of chars being used for extraction to prevent huge fields.
+| `fields`               | no        | all              | Properties to select to be stored, can be `content`, `title`, `name`, `author`, `keywords`, `date`, `content_type`, `content_length`, `language`
+|======
```

Comment:
I wonder if it would make sense for the ingest plugin to always fail the full ingest pipeline in case of parsing error? Should we ignore failures as we did with `ignore_errors` option. https://www.elastic.co/guide/en/elasticsearch/plugins/current/mapper-attachments-error-handling.html